modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/pagination/configuration/provider/PaginationConfigurationProvider.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.vulcan.internal.pagination.configuration.provider;
+
+import com.liferay.portal.configuration.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.dao.orm.QueryUtil;
+import com.liferay.portal.kernel.module.configuration.ConfigurationException;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.vulcan.internal.configuration.HeadlessAPICompanyConfiguration;
+import com.liferay.portal.vulcan.pagination.Pagination;
+
+import graphql.schema.DataFetchingEnvironment;
+
+import javax.servlet.http.HttpServletRequest;
+
+import javax.ws.rs.InternalServerErrorException;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Sergio Jiménez del Coso
+ */
+@Component(service = PaginationConfigurationProvider.class)
```

Comment:
`ConfigurationProvider` is a confusing term already used in Liferay from the configuration framework. The fact that you are using a configuration provider inside is just an implementation detail. I would call this simply `PaginationProvider`.